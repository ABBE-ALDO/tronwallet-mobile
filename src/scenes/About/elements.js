import styled, { css } from 'styled-components'
import { Colors } from '../../components/DesignSystem'

const BaseText = css`
  color: ${Colors.secondaryText};
  padding: 20px;
  align-self: center;
`
export const Description = styled.Text`
  font-family: Rubik-Light;
  font-size: 16px;
  line-height: 20px;

  color: white;
`
export const VersionText = styled.Text`
  font-family: Rubik-Regular;
  font-size: 14px;
  letter-spacing: 0.6;
  ${BaseText}
`
export const SectionTitle = styled.Text`
  margin-top: 5px;
  font-family: Rubik-Bold;
  font-size: 14px;
  ${BaseText}
`
export const PayPartner = styled.Image`
  width: 120px;
  height: 50px;
  resize-mode: contain;
`
export const Getty = styled.Image`
  width: 150px;
  height: 50px;
  resize-mode: contain;
`
const PaddingVertical = css`
  padding-vertical: 10px;
`
export const DescriptionWrapper = styled.View`
  ${PaddingVertical}
`
export const Partners = styled.View`
  ${PaddingVertical}
`
export const TutorialWrapper = styled.View`
  padding: 20px;
  background-color: ${Colors.secondaryText};
  border-radius: 4px;
`
export const TutorialText = styled.Text`
  font-family: Rubik-Bold;
  font-size: 14px;
  color: white;
  text-align: center;
`
