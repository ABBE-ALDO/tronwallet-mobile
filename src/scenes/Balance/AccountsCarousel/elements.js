import React from 'react'
import styled, { css } from 'styled-components'
import { View } from 'react-native'
import LinearGradient from 'react-native-linear-gradient'
import { Motion, spring } from 'react-motion'

import ClearButton from '../../../components/ClearButton'
import { Colors } from './../../../components/DesignSystem'
import * as Utils from './../../../components/Utils'

const borderRadius = css`
  border-radius: 6px;
`
export const CarouselCard = styled.View.attrs({
  elevation: 5
})`
  ${borderRadius} min-height: 180px;
  background-color: #66688f60;
  position: relative;
  margin-bottom: 10px;
  box-shadow: 0px 1px 1px rgba(0, 0, 0, 0.54);
`
export const TronLogo = styled.View`
  position: absolute;
  right: 15px;
  top: 20px;
  elevation: 5;
  width: 36px;
  height: 36px;
  border-radius: 18;
  box-shadow: 0px 1px 2px black;
`
export const Icon = styled.Image`
  height: 36px;
  width: 36px;
`
export const CardInfoText = styled.Text`
  font-family: 'Rubik-Medium';
  font-size: 11px;
  color: ${props => props.color};
`

CardInfoText.defaultProps = {
  color: Colors.primaryText
}

export const CardTopDecoration = styled(LinearGradient)`
  height: 3px;
  ${borderRadius};
`
export const CardBg = styled(LinearGradient)`
  flex: 1;
  padding: 22px;
  align-items: flex-start;
  ${borderRadius};
`

export const Stuff = styled.View`
  z-index: 1;
`

export const CardFooter = styled.View`
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  border-color: ${Colors.background};
  border-top-width: 1px;
  width: 100%;
  height: 41px;
  padding-left: 11px;
`

export const Value = styled.Text`
  font-family: 'Rubik-Medium';
  font-size: 16px;
  color: ${Colors.primaryText};
  ${props =>
    props.paddingRight &&
    css`
      padding-right: ${props.paddingRight}px;
    `};
`

export const Label = Value.extend`
  font-size: 12px;
  color: ${Colors.greyBlue};
  ${props =>
    props.paddingLeft &&
    css`
      padding-left: ${props.paddingLeft}px;
    `};
`

export const CardInfo = ({ label, value, currency, precision = 0 }) => (
  <View>
    <Label>{label}</Label>
    <Utils.VerticalSpacer size='xsmall' />
    <Motion defaultStyle={{ power: 0 }} style={{ power: spring(value) }}>
      {({ power }) => (
        <Value>{`${power.toFixed(precision)} ${currency}`}</Value>
      )}
    </Motion>
  </View>
)

export const BtnTrashWrapper = styled.View`
  width: 40px;
  height: 41px;
  border-left-width: 1px;
  border-color: ${Colors.background};
  align-content: center;
  align-items: center;
  justify-content: center;
`
export const BtnTrash = ({ onPress }) => (
  <BtnTrashWrapper>
    <ClearButton
      style={{ alignItems: 'center' }}
      onPress={onPress}
      padding={0}
      size={20}
    />
  </BtnTrashWrapper>
)

// borderTopLeftRadius: 6px;
// borderTopRightRadius: 6px;
