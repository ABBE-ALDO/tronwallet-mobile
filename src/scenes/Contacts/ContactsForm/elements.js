import styled from 'styled-components'

export const FormGroup = styled.View`
  margin-top: 35px;
`
export const CancelWrapper = styled.View`
  align-items: center;
`
