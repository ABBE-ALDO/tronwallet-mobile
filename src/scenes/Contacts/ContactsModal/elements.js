import styled, { css } from 'styled-components'
import { Colors } from '../../../components/DesignSystem'

export const ContactsModalWrapper = styled.TouchableHighlight`
  flex: 1;
  padding: 20px;
  background-color: rgba(0,0,0,0.5);
`
export const ContactsModalCard = styled.View`
  top: 150px;
  border-radius: 5px;
  background-color: ${Colors.dusk};
  padding-horizontal: 15px;
`
export const Divider = styled.View`
  height: 1px;
  background-color: ${Colors.background};
`
const Text = css`
  font-size: 12px;
  text-align: center;
`
export const Title = styled.Text`
  ${Text}
  font-family: Rubik-Medium;
  line-height: 16px;
  letter-spacing: 0;
  color: ${Colors.greyBlue};
  margin-top: 25px;
`
export const Action = styled.Text`
  ${Text}
  font-family: Rubik-Bold;
  line-height: 12px;
  letter-spacing: 0.6px;
  color: white;
  margin: 35px;
`
