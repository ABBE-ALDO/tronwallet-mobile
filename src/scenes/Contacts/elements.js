import styled, { css } from 'styled-components'
import { Colors } from '../../components/DesignSystem'

export const Card = styled.TouchableOpacity`
  border-left-width: 5px;
  border-color: #3f415d;
  height: 80px;
  margin-bottom: 1px;
  padding: 16px;
`
const Text = css`
  font-family: Rubik-Regular;
  line-height: 16px;
  color: ${Colors.greyBlue};
`
export const Name = styled.Text`
  ${Text}
  font-family: Rubik-Medium;
  font-size: 14px;
  letter-spacing: 0;
  color: white;
`
export const Alias = styled.Text`
  ${Text}
  font-size: 12px;
`
export const Address = styled.Text`
  ${Text}
  font-size: 11px;
`
