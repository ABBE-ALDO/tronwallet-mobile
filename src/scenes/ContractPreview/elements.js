import React, { PureComponent } from 'react'
import { TouchableOpacity } from 'react-native'
import styled from 'styled-components'
import { Text } from '../../components/Utils'
import ActionSheet from 'react-native-actionsheet'

const TouchableContainer = styled(TouchableOpacity)`
  flex: 1;
  justify-content: center;
  align-items: center;
`

export class AutoSignSelector extends PureComponent {
  _formatSignText = (option) => {
    if (!option.value) return option.text

    return `Auto sign will be enabled for ${option.text}`
  }

  render () {
    const { autoSign, options, onChange } = this.props

    return (
      <TouchableContainer onPress={() => this.ActionSheet.show()}>
        <Text size='smaller'>{this._formatSignText(autoSign)}</Text>
        <ActionSheet
          ref={ref => {
            this.ActionSheet = ref
          }}
          title={'Setting this option, all the trigger smart contracts comming from this address and calling that function will be automatically signed'}
          options={options.map(({text}) => text)}
          cancelButtonIndex={0}
          onPress={index => onChange(options[index])}
        />
      </TouchableContainer>
    )
  }
}
