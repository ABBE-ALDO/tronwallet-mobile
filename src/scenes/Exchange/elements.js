import React from 'react'
import styled from 'styled-components'
import { Colors, Spacing } from '../../components/DesignSystem'
import * as Utils from '../../components/Utils'

export const ExchangeRow = styled.TouchableOpacity`
  flex-direction: row;
  padding-horizontal: ${Spacing.medium};
  padding-vertical: ${Spacing.medium};
`
export const Divider = styled.View`
  width: 95%;
  background-color: ${Colors.greyBlue};
  height: 0.6;
`

export const TradePair = ({text}) =>
  <Utils.View align='center' justify='center'>
    <Utils.Text size='tiny' color={Colors.greyBlue}>
      {text}
    </Utils.Text>
  </Utils.View>
