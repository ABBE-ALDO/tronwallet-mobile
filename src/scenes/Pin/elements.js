import React from 'react'
import styled from 'styled-components'
import Ionicons from 'react-native-vector-icons/Ionicons'
import { transparentize } from 'polished'

import { Colors, Spacing } from '../../components/DesignSystem'

export const Label = styled.Text`
  font-family: Rubik-Medium;
  font-size: 11px;
  color: #66688F;
  letter-spacing: 0.6;
  line-height: 12px;
`

export const Text = styled.Text`
  font-family: Rubik-Medium;
  font-size: 16px;
  color: #FFF;
  line-height: 18px;
`

export const KeyPad = styled.View`
  flex-direction: row;
  flex-wrap: wrap;
  height: 242px;
  margin: 10px;
`

export const KeyWrapper = styled.TouchableOpacity`
  height: 52px;
  align-items: center;
  justify-content: center;
  flex-basis: 30%;
  flex-grow: 1;
  flex-shrink: 0;
  border-radius: 25px;
  border-width:${props => props.noBorder ? 0 : 1}px;
  border-color:${Colors.secondaryText};
  margin: 4px;
`

export const KeyText = styled.Text`
  font-family: Rubik-Regular;
  font-size: 16px;
  line-height: 24px;
  color:${Colors.secondaryText};
`

export const Key = ({ children, noBorder, onPress }) => (
  <KeyWrapper noBorder={noBorder} onPress={() => onPress(children)} >
    <KeyText>
      {children}
    </KeyText>
  </KeyWrapper>
)

export const Wrapper = styled.View`
  width: 36px;
  height: 36px;
  border-radius: 18px;
  align-items: center;
  justify-content: center;
  background-color: ${props => props.color};
`

Wrapper.defaultProps = {
  color: 'transparent'
}

export const Circle = styled.View`
  background-color: ${props => transparentize((1 - props.opacityAmount), props.color)};
  width: ${props => props.size}px;
  height: ${props => props.size}px;
  border-radius: ${props => props.size / 2}px;
  align-items: center;
  justify-content: center;
  z-index: 1;
`

Circle.defaultProps = {
  size: 12,
  opacityAmount: 1,
  color: '#2E2F47'
}

export const PinDigit = ({ digit, currentState }) => {
  if (digit < currentState.length) {
    return (
      <Wrapper>
        <Circle color='#9b9cb7' />
      </Wrapper>
    )
  }
  if (digit > currentState.length) {
    return (
      <Wrapper>
        <Circle />
      </Wrapper>
    )
  }
  return (
    <Wrapper>
      <Circle />
    </Wrapper>
  )
}

const GoBackBtn = styled.TouchableOpacity`
  position: absolute;
  top: 10;
  left: 8;
  padding: ${Spacing.medium}px;
  z-index: 1;
`

export const GoBackButton = props => (
  <GoBackBtn {...props} >
    <Ionicons name='ios-arrow-round-back' size={32} color='#FFF' />
  </GoBackBtn>
)

export const BiometricButton = styled.TouchableOpacity`
  padding: ${Spacing.medium}px;
`
