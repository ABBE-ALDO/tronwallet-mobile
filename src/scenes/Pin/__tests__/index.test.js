import React from 'react'
import 'react-native'
import PinScene from './../index'
import { BiometricButton } from './../elements'
import { shallow } from 'enzyme'
import 'jest-styled-components'

jest.mock('react-native-fabric')
jest.mock('react-native-mixpanel')
jest.mock('../../../utils/i18n')

jest.mock('react-native-vector-icons/Ionicons', () => jest.fn())

describe('Pin Scene', () => {
  const defaultProps = {
    context: { publicKey: '1234567890' }
  }
  const addListenerMockResult = jest.fn()
  const getParamMockResult = jest.fn()
  const addListenerMock = jest.fn(() => addListenerMockResult)
  const getParamMock = jest.fn(() => getParamMockResult)
  const props = {
    navigation: {
      addListener: addListenerMock,
      getParam: getParamMock,
      state: {
        testInput: jest.fn(),
        onSuccess: jest.fn()
      }
    }
  }

  const setup = (propOverrides = {}) => {
    const setupProps = { ...defaultProps, ...propOverrides }

    const wrapper = shallow(
      <PinScene {...setupProps} />
    )

    return {
      wrapper
    }
  }

  describe('componentDidMount', () => {
    test('#didFocus', () => {
      const { wrapper } = setup(props)
      const instance = wrapper.instance()
      instance.componentDidMount()
      expect(addListenerMock).toBeCalledWith('didFocus', instance._didFocus)
    })
  })

  describe('Biometrics', () => {
    test('Hide the biometrics button when #state.biometricsEnabled is false', () => {
      const { wrapper } = setup(props)
      const state = wrapper.state()
      expect(state).toMatchObject({ biometricsEnabled: false })
      expect(wrapper.find(BiometricButton).exists()).toBeFalsy()
    })

    test('I can see the biometrics button when #state.biometricsEnabled is true', () => {
      const { wrapper } = setup(props)
      const instance = wrapper.instance()
      instance.setState({ biometricsEnabled: true })
      const state = wrapper.state()
      expect(state).toMatchObject({ biometricsEnabled: true })
      expect(wrapper.find(BiometricButton).exists()).toBeTruthy()
    })

    test('Should clicking on the biometrics button when #state.biometricsEnabled is true', () => {
      const { wrapper } = setup(props)
      const instance = wrapper.instance()
      const spy = jest.spyOn(instance, '_handleBiometrics')
      instance.setState({ biometricsEnabled: true })
      const state = wrapper.state()
      const btnBio = wrapper.find(BiometricButton).first()
      expect(state).toMatchObject({ biometricsEnabled: true })
      btnBio.props().onPress()
      expect(spy).toHaveBeenCalled()
      spy.mockRestore()
    })
  })

  test('componentWillUnmount', () => {
    const backHandlerMock = jest.fn()
    const didFocusEvent = jest.fn()
    const { wrapper } = setup(props)
    const instance = wrapper.instance()
    instance.backHandler = { remove: backHandlerMock }
    instance.didFocusEvent = { remove: didFocusEvent }

    instance.componentWillUnmount()
    expect(backHandlerMock).toBeCalled()
    expect(didFocusEvent).toBeCalled()
  })

  test('Should match snapshot', () => {
    const { wrapper } = setup(props)
    expect(wrapper).toMatchSnapshot()
  })
})
