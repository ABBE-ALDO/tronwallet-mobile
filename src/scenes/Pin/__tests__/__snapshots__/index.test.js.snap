// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pin Scene Should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PinScene
    context={
      Object {
        "publicKey": "1234567890",
      }
    }
    navigation={
      Object {
        "addListener": [MockFunction] {
          "calls": Array [
            Array [
              "didFocus",
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": [MockFunction],
            },
          ],
        },
        "getParam": [MockFunction] {
          "calls": Array [
            Array [
              "shouldGoBack",
              false,
            ],
            Array [
              "shouldDoubleCheck",
              false,
            ],
            Array [
              "shouldGoBack",
              false,
            ],
            Array [
              "shouldDoubleCheck",
              false,
            ],
            Array [
              "shouldGoBack",
              false,
            ],
            Array [
              "shouldDoubleCheck",
              false,
            ],
            Array [
              "shouldGoBack",
              false,
            ],
            Array [
              "shouldDoubleCheck",
              false,
            ],
            Array [
              "shouldGoBack",
              false,
            ],
            Array [
              "shouldDoubleCheck",
              false,
            ],
            Array [
              "shouldGoBack",
              false,
            ],
            Array [
              "shouldDoubleCheck",
              false,
            ],
            Array [
              "shouldGoBack",
              false,
            ],
            Array [
              "shouldDoubleCheck",
              false,
            ],
            Array [
              "shouldGoBack",
              false,
            ],
            Array [
              "shouldDoubleCheck",
              false,
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": [MockFunction],
            },
            Object {
              "isThrow": false,
              "value": [MockFunction],
            },
            Object {
              "isThrow": false,
              "value": [MockFunction],
            },
            Object {
              "isThrow": false,
              "value": [MockFunction],
            },
            Object {
              "isThrow": false,
              "value": [MockFunction],
            },
            Object {
              "isThrow": false,
              "value": [MockFunction],
            },
            Object {
              "isThrow": false,
              "value": [MockFunction],
            },
            Object {
              "isThrow": false,
              "value": [MockFunction],
            },
            Object {
              "isThrow": false,
              "value": [MockFunction],
            },
            Object {
              "isThrow": false,
              "value": [MockFunction],
            },
            Object {
              "isThrow": false,
              "value": [MockFunction],
            },
            Object {
              "isThrow": false,
              "value": [MockFunction],
            },
            Object {
              "isThrow": false,
              "value": [MockFunction],
            },
            Object {
              "isThrow": false,
              "value": [MockFunction],
            },
            Object {
              "isThrow": false,
              "value": [MockFunction],
            },
            Object {
              "isThrow": false,
              "value": [MockFunction],
            },
          ],
        },
        "state": Object {
          "onSuccess": [MockFunction],
          "testInput": [MockFunction],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "background": "#191A2A",
      "children": <Styled(View)
        align="stretch"
        justify="flex-start"
      >
        <Styled(Styled(View))
          paddingSize="large"
        >
          <GoBackButton
            onPress={[Function]}
          />
          <Styled(View)
            align="center"
            justify="flex-start"
          >
            <Styled(Text)>
              SECURITY CHECK
            </Styled(Text)>
            <Styled(View)
              size="small"
            />
            <Styled(Text)>
              Enter New PIN
            </Styled(Text)>
          </Styled(View)>
        </Styled(Styled(View))>
        <PinPad
          onComplete={[Function]}
        />
      </Styled(View)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "align": "stretch",
        "children": Array [
          <Styled(Styled(View))
            paddingSize="large"
          >
            <GoBackButton
              onPress={[Function]}
            />
            <Styled(View)
              align="center"
              justify="flex-start"
            >
              <Styled(Text)>
                SECURITY CHECK
              </Styled(Text)>
              <Styled(View)
                size="small"
              />
              <Styled(Text)>
                Enter New PIN
              </Styled(Text)>
            </Styled(View)>
          </Styled(Styled(View))>,
          <PinPad
            onComplete={[Function]}
          />,
          null,
        ],
        "justify": "flex-start",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              false,
              <GoBackButton
                onPress={[Function]}
              />,
              <Styled(View)
                align="center"
                justify="flex-start"
              >
                <Styled(Text)>
                  SECURITY CHECK
                </Styled(Text)>
                <Styled(View)
                  size="small"
                />
                <Styled(Text)>
                  Enter New PIN
                </Styled(Text)>
              </Styled(View)>,
            ],
            "paddingSize": "large",
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onPress": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "align": "center",
                "children": Array [
                  <Styled(Text)>
                    SECURITY CHECK
                  </Styled(Text)>,
                  <Styled(View)
                    size="small"
                  />,
                  <Styled(Text)>
                    Enter New PIN
                  </Styled(Text)>,
                ],
                "justify": "flex-start",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "SECURITY CHECK",
                  },
                  "ref": null,
                  "rendered": "SECURITY CHECK",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "size": "small",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Enter New PIN",
                  },
                  "ref": null,
                  "rendered": "Enter New PIN",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onComplete": [Function],
          },
          "ref": [Function],
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "background": "#191A2A",
        "children": <Styled(View)
          align="stretch"
          justify="flex-start"
        >
          <Styled(Styled(View))
            paddingSize="large"
          >
            <GoBackButton
              onPress={[Function]}
            />
            <Styled(View)
              align="center"
              justify="flex-start"
            >
              <Styled(Text)>
                SECURITY CHECK
              </Styled(Text)>
              <Styled(View)
                size="small"
              />
              <Styled(Text)>
                Enter New PIN
              </Styled(Text)>
            </Styled(View)>
          </Styled(Styled(View))>
          <PinPad
            onComplete={[Function]}
          />
        </Styled(View)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": "stretch",
          "children": Array [
            <Styled(Styled(View))
              paddingSize="large"
            >
              <GoBackButton
                onPress={[Function]}
              />
              <Styled(View)
                align="center"
                justify="flex-start"
              >
                <Styled(Text)>
                  SECURITY CHECK
                </Styled(Text)>
                <Styled(View)
                  size="small"
                />
                <Styled(Text)>
                  Enter New PIN
                </Styled(Text)>
              </Styled(View)>
            </Styled(Styled(View))>,
            <PinPad
              onComplete={[Function]}
            />,
            null,
          ],
          "justify": "flex-start",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                false,
                <GoBackButton
                  onPress={[Function]}
                />,
                <Styled(View)
                  align="center"
                  justify="flex-start"
                >
                  <Styled(Text)>
                    SECURITY CHECK
                  </Styled(Text)>
                  <Styled(View)
                    size="small"
                  />
                  <Styled(Text)>
                    Enter New PIN
                  </Styled(Text)>
                </Styled(View)>,
              ],
              "paddingSize": "large",
            },
            "ref": null,
            "rendered": Array [
              false,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "onPress": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "align": "center",
                  "children": Array [
                    <Styled(Text)>
                      SECURITY CHECK
                    </Styled(Text)>,
                    <Styled(View)
                      size="small"
                    />,
                    <Styled(Text)>
                      Enter New PIN
                    </Styled(Text)>,
                  ],
                  "justify": "flex-start",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "SECURITY CHECK",
                    },
                    "ref": null,
                    "rendered": "SECURITY CHECK",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "size": "small",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Enter New PIN",
                    },
                    "ref": null,
                    "rendered": "Enter New PIN",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onComplete": [Function],
            },
            "ref": [Function],
            "rendered": null,
            "type": [Function],
          },
          null,
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenThreeAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
